src/Illuminate/Log/LogManager.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Illuminate\Log;
+
+use Psr\Log\AbstractLogger;
+use Psr\Log\LoggerInterface;
+use Illuminate\Contracts\Events\Dispatcher;
+
+class LogManager extends AbstractLogger implements LoggerInterface
+{
+    /**
+     * The available log channels.
+     *
+     * @var \Illuminate\Contracts\Log\Channel[]
+     */
+    protected $channels = [];
+
+    /**
+     * The default log channels.
+     *
+     * @var string[]
+     */
+    protected $defaultChannels = [];
+
+    /**
+     * The events dispatcher.
+     *
+     * @var \Illuminate\Contracts\Events\Dispatcher
+     */
+    protected $events;
+
+    /**
+     * Create the LogManager.
+     *
+     * @param  \Illuminate\Contracts\Events\Dispatcher  $events
+     */
+    public function __construct(Dispatcher $events)
+    {
+        $this->events = $events;
+    }
+
+    /**
+     * Retrieve a channel to use for logging.
+     *
+     * @param  string  $channel
+     * @return \Illuminate\Contracts\Log\Channel
+     */
+    public function to($channel)
+    {
+        return $this->resolveChannel($channel);
+    }
+
+    /**
+     * Attach an event listener to log a certain event.
+     *
+     * @param  string|array  $event
+     * @param  string  $level
+     * @param  null|array|string  $channels
+     * @return \Illuminate\Log\EventLogger|\Illuminate\Contracts\Log\Channel
+     */
+    public function event($event, $level = 'debug', $channels = null)
+    {
+        $logger = new EventLogger($this, $this->events, $event, $level);
+
+        if (! $channels) {
+            return $logger;
+        }
+
+        return $logger->to($channels);
+    }
+
+    /**
+     * Resolve a channel by name.
+     *
+     * @param  string  $channel
+     * @return \Illuminate\Contracts\Log\Channel
+     *
+     * @throws \Illuminate\Log\ChannelResolutionException
+     */
+    protected function resolveChannel($channel)
+    {
+        if (! array_key_exists($channel, $this->channels)) {
+            throw new ChannelResolutionException("Channel $channel is not registered.");
+        }
+
+        if ($this->channels[$channel] instanceof Channel) {
+            return $this->channels[$channel];
+        }
+
+        return $this->resolveChannel($this->channels[$channel]);
+    }
+
+    /**
+     * Register a new log channel.
+     *
+     * @param  string  $name
+     * @param  \Illuminate\Log\Channel  $channel
+     * @return void
+     */
+    public function registerChannel($name, Channel $channel)
```

Comment:
Shouldn't this be updated to take a chanel contract?

Reply:
Thanks Graham, good catch